addons/binding/org.openhab.binding.smappee/src/main/java/org/openhab/binding/smappee/internal/SmappeeService.java

```diff
@@ -0,0 +1,461 @@
+                "Mozilla/4.0 (compatible; MSIE 6.0; Windows NT 5.2; .NET CLR 1.0.3705;)");
+        postMethod.addRequestHeader("Accept", "application/json");
+
+        String accessTokenToInclude = GetAccessToken();
+        if (accessTokenToInclude.isEmpty()) {
+            logger.warn("Could not get access token");
+            return;
+        }
+
+        postMethod.addRequestHeader("Authorization", "Bearer " + accessTokenToInclude); // add the authorization header
+                                                                                        // to the request
+
+        try {
+            int statusCode = getClient.executeMethod(postMethod);
+            if (statusCode != HttpStatus.SC_OK) {
+                logger.warn("Post method failed: {}", postMethod.getStatusLine());
+                return;
+            }
+
+            return;
+        } catch (HttpException e) {
+            logger.warn("Fatal protocol violation: {}", e.toString());
+            return;
+        } catch (IOException e) {
+            logger.warn("Fatal transport error: {}", e.toString());
+            return;
+        } finally {
+            postMethod.releaseConnection();
+        }
+    }
+
+    private String GetAccessToken() {
+        if (accessToken != null && !accessToken.isEmpty() && accessTokenValidity != null
+                && accessTokenValidity.isBeforeNow()) {
+            return accessToken;
+        }
+
+        HttpClient postClient = new HttpClient();
+
+        if (accessTokenValidity != null) {
+            // get new accesstoken by using the refreshToken
+            PostMethod postMethod = new PostMethod("https://app1pub.smappee.net/dev/v1/oauth2/token");
+            postMethod.addRequestHeader("Accept", "application/json");
+
+            postMethod.addParameter("grant_type", "refresh_token");
+            postMethod.addParameter("refresh_token", refreshToken);
+            postMethod.addParameter("client_id", clientId);
+            postMethod.addParameter("client_secret", clientSecret);
+
+            try {
+                int statusCode = postClient.executeMethod(postMethod);
+                if (statusCode != HttpStatus.SC_OK) {
+                    logger.warn("Get Access Token failed: {}", postMethod.getStatusLine());
+                    return "";
+                }
+
+                String result = IOUtils.toString(postMethod.getResponseBodyAsStream());
+                SmappeeAccessTokenResponse accessTokenResponse = gson.fromJson(result,
+                        SmappeeAccessTokenResponse.class);
+
+                accessToken = accessTokenResponse.access_token;
+                refreshToken = accessTokenResponse.refresh_token;
+                accessTokenValidity = DateTime.now().plusSeconds(accessTokenResponse.expires_in);
+
+                return accessToken;
+
+            } catch (HttpException e) {
+                logger.warn("GetAccessToken on credentials : Fatal protocol violation: {}", e.toString());
```

Comment:
Added an extra test on the returnvalue of getData().

I agree that the user should look in the log. The exceptions that are logged here are operational errors, they should not occur often, but these errors should be logged.
Higher up the chain, the error is ignored, since it will be retried later on for the thing state updates. If the getdata() fails while initializing the thing, then the thing status is set to OFFLINE.